internal/internal_public.go

```diff
@@ -45,12 +45,13 @@ type (
 	// WorkflowTaskHandler represents decision task handlers.
 	WorkflowTaskHandler interface {
 		// Processes the workflow task
-		// The response is either
-		// RespondDecisionTaskCompletedRequest or RespondDecisionTaskFailedRequest
+		// The response could be:
+		// - RespondDecisionTaskCompletedRequest
+		// - RespondDecisionTaskFailedRequest
+		// - RespondQueryTaskCompletedRequest
 		ProcessWorkflowTask(
 			task *s.PollForDecisionTaskResponse,
-			historyIterator HistoryIterator,
-			emitStack bool) (response interface{}, stackTrace string, err error)
+			historyIterator HistoryIterator) (response interface{}, context *WorkflowExecutionContext, err error)
```

Comment:
After this change WorkflowExecutionContext plays dual role. One for workflow implementation code, another for poller. I think returning an interface from this call with well defined set of methods would make it cleaner. It is OK if the current context does implement this interface. Then workflowExecutionContext might stay private.

Reply:
I can do a follow up refactoring change on this.